packages/@sanity/block-tools/src/util/normalizeBlock.ts

```diff
@@ -30,6 +30,11 @@ export default function normalizeBlock(block) {
     return block
   }
   let usedMarkDefs = []
+  const allowedDecorators =
+    options.decorators &&
+    Array.isArray(options.decorators) &&
+    options.decorators.length > 0 &&
```

Comment:
If you have a block where no decorators are allowed, doesn't this check prevent you from stripping disallowed decorators?

Reply:
Very good point!